clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -3036,12 +3178,10 @@ public void testEpochUpdateAfterBumpFromEndTxnResponseInV2() throws InterruptedE
         doInitTransactions(producerId, epoch);
 
         transactionManager.beginTransaction();
-        transactionManager.maybeAddPartition(tp0);
-        prepareAddPartitionsToTxnResponse(Errors.NONE, tp0, epoch, producerId);
-        runUntil(() -> transactionManager.isPartitionAdded(tp0));
 
         // Append record with initial producer ID and epoch.
         Future<RecordMetadata> responseFuture = appendToAccumulator(tp0);
+        transactionManager.maybeAddPartition(tp0);
```

Comment:
Hmmm -- was the ordering before incorrect? I noticed there was a comment we have to add after adding to the accumulator.

Reply:
It order does not matter, but I think I should move it back.